### PR TITLE
fix: [PL-42825]: make sorting users on email and name case insensitive

### PR DESCRIPTION
**Pull request by sahibaHarness**
## Describe your changes
image

Index Utilization
image

Tested for:
-> Tested for sorting orders supported on UI
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55676)

### CODE REVIEW

**Comment by sahibaHarness** [comment]
trigger messagecheck

**Comment by sahibaHarness** [comment]
trigger uts

**Comment by sahibaHarness** [comment]
trigger uts

File: 120-ng-manager/src/main/java/io/harness/repositories/invites/custom/InviteRepositoryCustomImpl.java [hunk 1]
```diff
@@ -61,4 +62,13 @@ public DeleteResult deleteAll(Criteria criteria) {
     Query query = new Query(criteria);
     return mongoTemplate.remove(query, Invite.class);
   }
+
+  @Override
+  public Page<Invite> findAllWithCollation(Criteria criteria, Pageable pageable) {
+    Query query = new Query(criteria).with(pageable);
+    query.collation(Collation.of("en").strength(Collation.ComparisonLevel.secondary()));
```
Comment (tiwaritejas) [inline 120-ng-manager/src/main/java/io/harness/repositories/invites/custom/InviteRepositoryCustomImpl.java:69]: @sahibaHarness Can you check how does this affect the index usage? There have been incidents in the past where making a query insensitive led to poor index usage, thereby slowing down the query and increase mongo CPU load.

**Review by tiwaritejas — CHANGES_REQUESTED** [review]

File: 120-ng-manager/src/main/java/io/harness/repositories/invites/custom/InviteRepositoryCustomImpl.java (hunk 1 above)
Comment (sahibaHarness) [reply 120-ng-manager/src/main/java/io/harness/repositories/invites/custom/InviteRepositoryCustomImpl.java:69]: image
I had a discussion on these indexes with @anurag as well on the indexes


`{
    "queryPlanner" : {
        "plannerVersion" : 1.0,
        "namespace" : "ng-harness.invites",
        "indexFilterSet" : false,
        "parsedQuery" : {
            "$and" : [
                {
                    "accountIdentifier" : {
                        "$eq" : "kmpySmUISimoRrJL6NL73w"
                    }
                },
                {
                    "deleted" : {
                        "$eq" : false
                    }
                },
                {
                    "orgIdentifier" : {
                        "$eq" : null
                    }
                },
                {
                    "projectIdentifier" : {
                        "$eq" : null
                    }
                }
            ]
        },
        "collation" : {
            "locale" : "en",
            "caseLevel" : false,
            "caseFirst" : "off",
            "strength" : 2.0,
            "numericOrdering" : false,
            "alternate" : "non-ignorable",
            "maxVariable" : "punct",
            "normalization" : false,
            "backwards" : false,
            "version" : "57.1"
        },
        "winningPlan" : {
            "stage" : "FETCH",
            "filter" : {
                "$and" : [
                    {
                        "orgIdentifier" : {
                            "$eq" : null
                        }
                    },
                    {
                        "projectIdentifier" : {
                            "$eq" : null
                        }
                    }
                ]
            },
            "inputStage" : {
                "stage" : "IXSCAN",
                "keyPattern" : {
                    "deleted" : 1.0,
                    "name" : 1.0,
                    "accountIdentifier" : 1.0,
                    "orgIdentifier" : 1.0,
                    "projectIdentifier" : 1.0,
                    "roleBindings" : 1.0
                },
                "indexName" : "ng_invite_deleted_name_account_org_project_roleBindings_with_collation",
                "collation" : {
                    "locale" : "en",
                    "caseLevel" : false,
                    "caseFirst" : "off",
                    "strength" : 2.0,
                    "numericOrdering" : false,
                    "alternate" : "non-ignorable",
                    "maxVariable" : "punct",
                    "normalization" : false,
                    "backwards" : false,
                    "version" : "57.1"
                },
                "isMultiKey" : true,
                "multiKeyPaths" : {
                    "deleted" : [

                    ],
                    "name" : [

                    ],
                    "accountIdentifier" : [

                    ],
                    "orgIdentifier" : [

                    ],
                    "projectIdentifier" : [

                    ],
                    "roleBindings" : [
                        "roleBindings"
                    ]
                },
                "isUnique" : false,
                "isSparse" : false,
                "isPartial" : false,
                "indexVersion" : 2.0,
                "direction" : "backward",
                "indexBounds" : {
                    "deleted" : [
                        "[false, false]"
                    ],
                    "name" : [
                        "[MaxKey, MinKey]"
                    ],
                    "accountIdentifier" : [
                        "[\"=AGYMAQ9M9AEKK;?\\x1EC? \\x18U\\x01\\x1A\", \"=AGYMAQ9M9AEKK;?\\x1EC? \\x18U\\x01\\x1A\"]"
                    ],
                    "orgIdentifier" : [
                        "[null, null]",
                        "[undefined, undefined]"
                    ],
                    "projectIdentifier" : [
                        "[null, null]",
                        "[undefined, undefined]"
                    ],
                    "roleBindings" : [
                        "[MaxKey, MinKey]"
                    ]
                }
            }
        },
        "rejectedPlans" : [
            {
                "stage" : "SORT",
                "sortPattern" : {
                    "name" : -1.0
                },
                "inputStage" : {
                    "stage" : "SORT_KEY_GENERATOR",
                    "inputStage" : {
                        "stage" : "FETCH",
                        "filter" : {
                            "$and" : [
                                {
                                    "orgIdentifier" : {
                                        "$eq" : null
                                    }
                                },
                                {
                                    "projectIdentifier" : {
                                        "$eq" : null
                                    }
                                },
                                {
                                    "accountIdentifier" : {
                                        "$eq" : "kmpySmUISimoRrJL6NL73w"
                                    }
                                }
                            ]
                        },
                        "inputStage" : {
                            "stage" : "IXSCAN",
                            "keyPattern" : {
                                "deleted" : 1.0,
                                "email" : 1.0,
                                "accountIdentifier" : 1.0,
                                "orgIdentifier" : 1.0,
                                "projectIdentifier" : 1.0
                            },
                            "indexName" : "ng_invite_account_org_project_identifiers_email_role_deleted",
                            "isMultiKey" : false,
                            "multiKeyPaths" : {
                                "deleted" : [

                                ],
                                "email" : [

                                ],
                                "accountIdentifier" : [

                                ],
                                "orgIdentifier" : [

                                ],
                                "projectIdentifier" : [

                                ]
                            },
                            "isUnique" : false,
                            "isSparse" : false,
                            "isPartial" : false,
                            "indexVersion" : 2.0,
                            "direction" : "forward",
                            "indexBounds" : {
                                "deleted" : [
                                    "[false, false]"
                                ],
                                "email" : [
                                    "[MinKey, MaxKey]"
                                ],
                                "accountIdentifier" : [
                                    "[MinKey, MaxKey]"
                                ],
                                "orgIdentifier" : [
                                    "[undefined, undefined]",
                                    "[null, null]"
                                ],
                                "projectIdentifier" : [
                                    "[undefined, undefined]",
                                    "[null, null]"
                                ]
                            }
                        }
                    }
                }
            },
            {
                "stage" : "SORT",
                "sortPattern" : {
                    "name" : -1.0
                },
                "inputStage" : {
                    "stage" : "SORT_KEY_GENERATOR",
                    "inputStage" : {
                        "stage" : "FETCH",
                        "filter" : {
                            "$and" : [
                                {
                                    "orgIdentifier" : {
                                        "$eq" : null
                                    }
                                },
                                {
                                    "projectIdentifier" : {
                                        "$eq" : null
                                    }
                                }
                            ]
                        },
                        "inputStage" : {
                            "stage" : "IXSCAN",
                            "keyPattern" : {
                                "deleted" : 1.0,
                                "email" : 1.0,
                                "accountIdentifier" : 1.0,
                                "orgIdentifier" : 1.0,
                                "projectIdentifier" : 1.0,
                                "roleBindings" : 1.0
                            },
                            "indexName" : "ng_invite_deleted_email_account_org_project_roleBindings_with_collation",
                            "collation" : {
                                "locale" : "en",
                                "caseLevel" : false,
                                "caseFirst" : "off",
                                "strength" : 2.0,
                                "numericOrdering" : false,
                                "alternate" : "non-ignorable",
                                "maxVariable" : "punct",
                                "normalization" : false,
                                "backwards" : false,
                                "version" : "57.1"
                            },
                            "isMultiKey" : true,
                            "multiKeyPaths" : {
                                "deleted" : [

                                ],
                                "email" : [

                                ],
                                "accountIdentifier" : [

                                ],
                                "orgIdentifier" : [

                                ],
                                "projectIdentifier" : [

                                ],
                                "roleBindings" : [
                                    "roleBindings"
                                ]
                            },
                            "isUnique" : false,
                            "isSparse" : false,
                            "isPartial" : false,
                            "indexVersion" : 2.0,
                            "direction" : "forward",
                            "indexBounds" : {
                                "deleted" : [
                                    "[false, false]"
                                ],
                                "email" : [
                                    "[MinKey, MaxKey]"
                                ],
                                "accountIdentifier" : [
                                    "[\"=AGYMAQ9M9AEKK;?\\x1EC? \\x18U\\x01\\x1A\", \"=AGYMAQ9M9AEKK;?\\x1EC? \\x18U\\x01\\x1A\"]"
                                ],
                                "orgIdentifier" : [
                                    "[undefined, undefined]",
                                    "[null, null]"
                                ],
                                "projectIdentifier" : [
                                    "[undefined, undefined]",
                                    "[null, null]"
                                ],
                                "roleBindings" : [
                                    "[MinKey, MaxKey]"
                                ]
                            }
                        }
                    }
                }
            }
        ]
    },
    "executionStats" : {
        "executionSuccess" : true,
        "nReturned" : 3.0,
        "executionTimeMillis" : 3.0,
        "totalKeysExamined" : 4.0,
        "totalDocsExamined" : 3.0,
        "executionStages" : {
            "stage" : "FETCH",
            "filter" : {
                "$and" : [
                    {
                        "orgIdentifier" : {
                            "$eq" : null
                        }
                    },
                    {
                        "projectIdentifier" : {
                            "$eq" : null
                        }
                    }
                ]
            },
            "nReturned" : 3.0,
            "executionTimeMillisEstimate" : 0.0,
            "works" : 6.0,
            "advanced" : 3.0,
            "needTime" : 1.0,
            "needYield" : 0.0,
            "saveState" : 0.0,
            "restoreState" : 0.0,
            "isEOF" : 1.0,
            "docsExamined" : 3.0,
            "alreadyHasObj" : 0.0,
            "inputStage" : {
                "stage" : "IXSCAN",
                "nReturned" : 3.0,
                "executionTimeMillisEstimate" : 0.0,
                "works" : 5.0,
                "advanced" : 3.0,
                "needTime" : 1.0,
                "needYield" : 0.0,
                "saveState" : 0.0,
                "restoreState" : 0.0,
                "isEOF" : 1.0,
                "keyPattern" : {
                    "deleted" : 1.0,
                    "name" : 1.0,
                    "accountIdentifier" : 1.0,
                    "orgIdentifier" : 1.0,
                    "projectIdentifier" : 1.0,
                    "roleBindings" : 1.0
                },
                "indexName" : "ng_invite_deleted_name_account_org_project_roleBindings_with_collation",
                "collation" : {
                    "locale" : "en",
                    "caseLevel" : false,
                    "caseFirst" : "off",
                    "strength" : 2.0,
                    "numericOrdering" : false,
                    "alternate" : "non-ignorable",
                    "maxVariable" : "punct",
                    "normalization" : false,
                    "backwards" : false,
                    "version" : "57.1"
                },
                "isMultiKey" : true,
                "multiKeyPaths" : {
                    "deleted" : [

                    ],
                    "name" : [

                    ],
                    "accountIdentifier" : [

                    ],
                    "orgIdentifier" : [

                    ],
                    "projectIdentifier" : [

                    ],
                    "roleBindings" : [
                        "roleBindings"
                    ]
                },
                "isUnique" : false,
                "isSparse" : false,
                "isPartial" : false,
                "indexVersion" : 2.0,
                "direction" : "backward",
                "indexBounds" : {
                    "deleted" : [
                        "[false, false]"
                    ],
                    "name" : [
                        "[MaxKey, MinKey]"
                    ],
                    "accountIdentifier" : [
                        "[\"=AGYMAQ9M9AEKK;?\\x1EC? \\x18U\\x01\\x1A\", \"=AGYMAQ9M9AEKK;?\\x1EC? \\x18U\\x01\\x1A\"]"
                    ],
                    "orgIdentifier" : [
                        "[null, null]",
                        "[undefined, undefined]"
                    ],
                    "projectIdentifier" : [
                        "[null, null]",
                        "[undefined, undefined]"
                    ],
                    "roleBindings" : [
                        "[MaxKey, MinKey]"
                    ]
                },
                "keysExamined" : 4.0,
                "seeks" : 2.0,
                "dupsTested" : 3.0,
                "dupsDropped" : 0.0
            }
        }
    },
    "serverInfo" : {
        "host" : "0c6b4dead198",
        "port" : 27017.0,
        "version" : "4.2.24",
        "gitVersion" : "5e4ec1d24431fcdd28b579a024c5c801b8cde4e2"
    },
    "ok" : 1.0
}
`

**Review by adiyaar24 — APPROVED** [review]

**Review by tiwaritejas — APPROVED** [review]

**Review by bhavya181 — APPROVED** [review]

**Pull request merged**